sunpy/util/util.py

```diff
@@ -212,3 +212,38 @@ def expand_list_generator(inp):
                 yield nested_item
         else:
             yield item
+
+def partial_match(key, dictionary):
+    """
+    Return value/values based on the partial key.
+    Each element of the partial key is matched against the keys of the dictionary and if a match is found 
+    the value of the key is returned.
+    Even a partial match works here i.e even if the key matches partially a value is returned.
+    Note that this function uses a generator so we need to type cast the output to a list to get the values.
```

Comment:
You get values via a generator too! That's only if you want to get them all at once.

Reply:
Here we need all the values so that is why I added that line. I will change it and make it say if `if you need to get all the values`

Reply:
I don't think these notes should be here. I understand they can be useful for beginners, but there should be other place to put that. You don't get that kind of information when you look at the docstring of `range` or `map`. They say "Make an iterator...", then the person can look up how an `iterator` works. Here you could say, it returns a `generator`.